### PR TITLE
Julia 1.4.1

### PR DESCRIPTION
**Pull request by ivanslapnicar**
Syntax changes so that the package works on Julia 1.4.1 (and probably on 1.1, 1.3). Also, unzip was changes to InfoZIP.jl.

### CODE REVIEW

File: src/IncrementalSVD.jl [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module IncrementalSVD
 
-using ProgressMeter
+using ProgressMeter, InfoZIP, Random, LinearAlgebra
```
Comment (aaw) [inline src/IncrementalSVD.jl:3]: I understand why you added InfoZip here, but why add Random and LinearAlgebra?

File: src/movielens_dataset.jl [hunk 2]
```diff
@@ -8,7 +8,7 @@
 # Both datasets have ratings in the range 1.0-5.0.
 
 function load_movielens_dataset(zipfile_name, archive_dir, split_ratio=0.10)
-    temp_dir = tempdir()
+    temp_dir = "." # tempdir()
```
Comment (aaw) [inline src/movielens_dataset.jl]: I'd like to keep tempdir() here. Can you revert this line or add some way of overriding it if you need a directory other than tempdir() (like passing a parameter temp_dir=nothing that is set to tempdir() if nothing is passed)?

File: src/movielens_dataset.jl [hunk 3]
```diff
@@ -20,11 +20,7 @@ function load_movielens_dataset(zipfile_name, archive_dir, split_ratio=0.10)
     ratings_file_name = joinpath(temp_dir, archive_dir, "ratings.dat")
     if !isfile(movie_file_name)
         println("Extracting movie data...")
-        run(`unzip $download_path $(archive_dir)/movies.dat -d $temp_dir`)
```
Comment (aaw) [inline src/movielens_dataset.jl:23]: It looks like you're removing lines 23-27 when I think you should be just replacing line 23 with "InfoZIP.unzip(...". Can you fix?

File: src/movielens_dataset.jl [hunk 4]
```diff
@@ -41,6 +37,7 @@ function load_movielens_dataset(zipfile_name, archive_dir, split_ratio=0.10)
     user_to_index = Dict{AbstractString, Int32}()
     ratings_count = countlines(ratings_file_name)
     progress = Progress(ratings_count, 1, "Loading ratings ")
+    println(ratings_file_name)
```
Comment (aaw) [inline src/movielens_dataset.jl]: Can you remove this println?

**Comment by ivanslapnicar** [comment]
Dear Aaaron,

thanks for your letter. Random is needed for shuffle(), LinearAlgebra is needed for norm().

Best, Ivan


Prof. dr. sc. Ivan Slapničar
Sveučilište u Splitu / University of Split
FESB
R. Boškovića 32
HR-21000 Split
Hrvatska / Croatia
Phone: + 385 21 305893
Ured/Office: B803

________________________________
Šalje: Aaron Windsor <notifications@github.com>
Poslano: 3. svibnja 2020. 18:02
Prima: aaw/IncrementalSVD.jl <IncrementalSVD.jl@noreply.github.com>
Kopija: Ivan Slapničar <slap@fesb.hr>; Author <author@noreply.github.com>
Predmet: Re: [aaw/IncrementalSVD.jl] Julia 1.4.1 (#12)


@aaw commented on this pull request.

________________________________

In src/IncrementalSVD.jl<https://github.com/aaw/IncrementalSVD.jl/pull/12#discussion_r419123365>:

> @@ -1,6 +1,6 @@
 module IncrementalSVD

-using ProgressMeter
+using ProgressMeter, InfoZIP, Random, LinearAlgebra


I understand why you added InfoZip here, but why add Random and LinearAlgebra?

**Comment by ivanslapnicar** [comment]
Keeping tempdir is no problem.

Prof. dr. sc. Ivan Slapničar
Sveučilište u Splitu / University of Split
FESB
R. Boškovića 32
HR-21000 Split
Hrvatska / Croatia
Phone: + 385 21 305893
Ured/Office: B803

________________________________
Šalje: Aaron Windsor <notifications@github.com>
Poslano: 3. svibnja 2020. 18:02
Prima: aaw/IncrementalSVD.jl <IncrementalSVD.jl@noreply.github.com>
Kopija: Ivan Slapničar <slap@fesb.hr>; Author <author@noreply.github.com>
Predmet: Re: [aaw/IncrementalSVD.jl] Julia 1.4.1 (#12)


@aaw commented on this pull request.

________________________________

In src/movielens_dataset.jl<https://github.com/aaw/IncrementalSVD.jl/pull/12#discussion_r419123376>:

> @@ -8,7 +8,7 @@
 # Both datasets have ratings in the range 1.0-5.0.

 function load_movielens_dataset(zipfile_name, archive_dir, split_ratio=0.10)
-    temp_dir = tempdir()
+    temp_dir = "." # tempdir()


I'd like to keep tempdir() here. Can you revert this line or add some way of overriding it if you need a directory other than tempdir() (like passing a parameter temp_dir=nothing that is set to tempdir() if nothing is passed)?

**Comment by ivanslapnicar** [comment]
In the original, unzip is called twice, once for each file.
InfoZIP.unzi extracts the entire zip file at once.
Does that make sense?



Prof. dr. sc. Ivan Slapničar
Sveučilište u Splitu / University of Split
FESB
R. Boškovića 32
HR-21000 Split
Hrvatska / Croatia
Phone: + 385 21 305893
Ured/Office: B803

________________________________
Šalje: Aaron Windsor <notifications@github.com>
Poslano: 3. svibnja 2020. 18:02
Prima: aaw/IncrementalSVD.jl <IncrementalSVD.jl@noreply.github.com>
Kopija: Ivan Slapničar <slap@fesb.hr>; Author <author@noreply.github.com>
Predmet: Re: [aaw/IncrementalSVD.jl] Julia 1.4.1 (#12)


@aaw commented on this pull request.

________________________________

In src/movielens_dataset.jl<https://github.com/aaw/IncrementalSVD.jl/pull/12#discussion_r419123396>:

> @@ -20,11 +20,7 @@ function load_movielens_dataset(zipfile_name, archive_dir, split_ratio=0.10)
     ratings_file_name = joinpath(temp_dir, archive_dir, "ratings.dat")
     if !isfile(movie_file_name)
         println("Extracting movie data...")
-        run(`unzip $download_path $(archive_dir)/movies.dat -d $temp_dir`)


It looks like you're removing lines 23-27 when I think you should be just replacing line 23 with "InfoZIP.unzip(...". Can you fix?

**Comment by ivanslapnicar** [comment]
Sure!


Prof. dr. sc. Ivan Slapničar
Sveučilište u Splitu / University of Split
FESB
R. Boškovića 32
HR-21000 Split
Hrvatska / Croatia
Phone: + 385 21 305893
Ured/Office: B803

________________________________
Šalje: Aaron Windsor <notifications@github.com>
Poslano: 3. svibnja 2020. 18:02
Prima: aaw/IncrementalSVD.jl <IncrementalSVD.jl@noreply.github.com>
Kopija: Ivan Slapničar <slap@fesb.hr>; Author <author@noreply.github.com>
Predmet: Re: [aaw/IncrementalSVD.jl] Julia 1.4.1 (#12)


@aaw commented on this pull request.

________________________________

In src/movielens_dataset.jl<https://github.com/aaw/IncrementalSVD.jl/pull/12#discussion_r419123411>:

> @@ -41,6 +37,7 @@ function load_movielens_dataset(zipfile_name, archive_dir, split_ratio=0.10)
     user_to_index = Dict{AbstractString, Int32}()
     ratings_count = countlines(ratings_file_name)
     progress = Progress(ratings_count, 1, "Loading ratings ")
+    println(ratings_file_name)


Can you remove this println?

**Comment by aaw** [comment]
Thanks for the fixes!

**Comment by ivanslapnicar** [comment]
You are welcome. It was a pleasure! It is a great package. I am using it in my Applied NLA Course.

**Pull request merged**